### DISCUSSION TITLE
PrimeVue DataTable unable to use SHIFT to select multiple rows if checkbox mode is enabled.

### DISCUSSION THREAD

**Discussion by redfox05** [primefaces/community#2204] [PrimeVue] [answered]
Without checkbox, you seem to be able to select multiple rows using SHIFT + Click.
But not on ones with the checkbox column. This is using the examples at https://primevue.org/datatable/#multiple_rows_selection and https://primevue.org/datatable/#checkbox_row_selection

Can someone clarify if this really is not possible to do, or did I just miss a config flag somewhere that was different between the two example tables?

Alternatively, is there a way to programatically decide which rows to select? Then I can write my own SHIFT + Click code to decide which rows to select.

Would appreciate if someone could point me in the right direction as it's taken quite a bit of time to trial and error this without success.

**Comment by redfox05** [comment, marked as answer]
Found the solution, I was ready to start implementing this and potentially submit a MR, but turns out it was supported all along, it was just not well described in the documentation. Turns out `selectionMode="multiple"` has to be defined on the checkbox column AND the table. Given that they have the same name, the different use cases of this multiple selection mode could be described better in the docs. I'm happy to provide a documentation MR for this if the team wants.

And for anyone else coming across this that wants to further fine tune the selection process, if you want a user to be able to reset the selection process when they select another record after using SHIFT to select a range, then you want to add `:metaKeySelection="true"` to the table definition. Or use false (or omit completely) if you want further clicks of rows to add to the already selected rows, rather than clear them. Note that click behaviour on rows is different to click behaviour on checkboxes.

**Reply by zuffobeemon** [reply]
I overwritten the `DataTableRowClickEvent` for open a dialog, so when using `selectionMode="multiple"` on `<Datatable>`, the SHIFT + Click works clicking on the row and does not work when clicking on the select input from `<Column selectionMode="multiple" />`.

Is there a way to enable SHIFT + Click on the select input from `<Column selectionMode="multiple" />` and disable any other place?

**Comment by tommasogangemi** [comment]
@redfox05 I read and tried your solution but it doesn't look like it's working. I did the following:

- went on the documentation page
- opened the check box selection example in stackblitz
- added the `selectionMode="multiple"` prop also to the DataTable component

And it does't look to be working, is it what you suggested as the solution, or am I missing something? Here is [the link](https://stackblitz.com/edit/ajbylk9o?file=src%2FApp.vue) to the stackblitz, in case someone wants to check it out.

**Reply by tommasogangemi** [reply]
Actually let me rectify. The proposed solution works, when the elements I click on are the rows and not the checkboxes, which means that the feature you are turning on is not shift-key checkbox selection, but really just multiple row selection as in the first piece of doc you linked. 

Using checkboxes is still not working at all.

**Reply by Quent1OK** [reply]
The solution has been implemented in this [Pull Request](https://github.com/primefaces/primevue/pull/8349) and is now awaiting review by the PrimeVue team.
